src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -215,9 +217,128 @@ public SAMSequenceRecord addSequenceAlias(final String originalName,
     }
-
+    
+    /** 
+     * called by `parseInterval`. Parse a coordinate string, remove the comma look at the suffix (bp,kb,mp,gb) to scale,
+     * verify that the result is a positive integer
+     * @author Pierre Lindenbaum 
+     * @param numStr the coordinate
+     * @return the coordinate as integer
+     */
+    private static int parseCoordinate(final String numStr)
+        {
+        long factor = 1L;
+        String num = numStr.replace(",","").trim().toLowerCase();
+        if(num.endsWith("bp"))
+            {
+            factor = 1L;
+            num = num.substring(0,num.length()-2).trim();
+            }
+        else if(num.endsWith("kb"))
+            {
+            factor = 1000L;//1E3
+            num = num.substring(0,num.length()-2).trim();
+            }
+        else if(num.endsWith("mb"))
+            {
+            factor = 1000000L;//1E6
+            num = num.substring(0,num.length()-2).trim();
+            }
+        else if(num.endsWith("gb"))
+            {
+            factor = 1000000000L;//1E9
+            num = num.substring(0,num.length()-2).trim();
+            }
+        
+        try {
+            long pos = Long.parseLong(num);
+            pos *= factor;
+            if(pos < 1L || pos > Integer.MAX_VALUE)  throw new IllegalArgumentException("Cannot convert "+numStr+" to a positive integer");
+            return (int)pos;
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Cannot convert "+numStr+" to a coordinate");
+            }
+        }
+    
+    /** 
+     * parses the interval String and returns a htsjdk.samtools.util.Interval
+     * The contig/chromosome must exist in the dictionary. out-of-bound index <code>!(1<= index <= sequence.length())</code> will be clipped 
+     * coordinate suffixes like 'bp,Mb,Gb' will be interpreted.
+     * Comma in the coordinate will be removed.
+     * left/right white spaces will be trimmed.
+     * Position=0 will be set to 1.
+     * A Position larger than the contig size will be set to the contig size
+     * 
+     * Valid syntax -> convertion :
+     * <pre> 
+     *     CHROM -> CHROM:1-seq.length()
+     *     CHROM:POS ->  CHROM:POS-POS
+     *     CHROM:POS+NUM -> CHROM:{POS-NUM}-{POS+NUM}
+     *     CHROM:BEGIN- ->  CHROM:BEGIN-seq.length()
+     *     CHROM:BEGIN-END ->  CHROM:BEGIN-END
+     * </pre>
+     * @author Pierre Lindenbaum 
+     * @param region a 1-based coordinate region.  
+     * @return the parsed htsjdk.samtools.util.Interval
+     */
+    public Interval parseInterval(final String region) {
+        if( region == null) throw new IllegalArgumentException("parseInterval: region is null");
+        if(region.isEmpty())  throw new IllegalArgumentException("parseInterval: region is empty");
+        final int colon = region.indexOf(':');
+        // no colon, chromosome only
+        if(colon == -1) {
+            final SAMSequenceRecord record = getSequence(region);
```

Comment:
There's a lot of parsing going on here. Would it be possible to use a regex to capture the parsing? 
something like this:
```code
^(.*?)((:)(.*?)(([-+])?(.*?))?)?$
```

Then the first group would be the CHROM
if there is a second group, the fourth would be POS1 the sixth (if there) would be the + or -  and the seventh (if there) would the POS2, then you can separate the parsing from the logic of constructing the interval, which currently is combined in a complicated way.

I'm not saying it has to be my regex...that is just an example...but I think it would be good to separate the parsing of the string and the logic of creating the Interval.